edk2toolext/tests/test_edk2_update.py

```diff
@@ -0,0 +1,117 @@
+# @file test_edk2_update.py
+# This contains unit tests for the edk2_update
+##
+# Copyright (c) Microsoft Corporation
+#
+# SPDX-License-Identifier: BSD-2-Clause-Patent
+##
+import unittest
+from edk2toolext.invocables.edk2_update import Edk2Update
+import tempfile
+import sys
+import os
+import logging
+from importlib import reload
+from edk2toolext.environment import shell_environment
+from edk2toolext.environment import self_describing_environment
+
+class_info = '''
+from edk2toolext.invocables.edk2_update import UpdateSettingsManager
+class TestSettingsManager(UpdateSettingsManager):
+
+    def GetActiveScopes(self):
+        return []
+
+    def GetWorkspaceRoot(self):
+        return WORKSPACE
+
+    def AddCommandLineOptions(self, parserObj):
+        pass
+
+    def RetrieveCommandLineOptions(self, args):
+        pass
+
+    def GetName(self):
+        return "TestUpdate"
+
+    def GetArchitecturesSupported(self):
+        return []
+
+    def GetPackagesSupported(self):
+        return []
+
+    def GetTargetsSupported(self):
+        return []
+'''
+
+# TODO: move to something lighter weight that has external dependencies
+# Perhaps create a custom nuget package that has itself in it?
+basecore_ext_dep = '''
```

Comment:
i think you should move to the octocat sample repo or if you need ext_deps in it we can make a test repo based off octocat.  Mu Basecore has got to waste a lot of time and make the unit test not so light weight.

Reply:
Agreed- either an octocat sample repo or a nuget package.